### ISSUE TITLE
new lint: sort collection after checking it is empty

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#3914]
````rust
let mut v: Vec<_> = ...
v.sort(); // move into condition
if v.is_empty() {
    bla();
} else {
    foo(&v);
}
````
There is no need to sort `v` if we know that it is empty, could we suggest sinking the sort() into the `else` branch?

````rust
if v.is_empty() {
    bla();
} else {
    v.sort(); // moved here
    foo(&v);
}
````

**Comment by matthiaskrgr** [comment]
If we see a `.sort()` after asserting `.is_empty()`, this might even be a `correctness` lint? 
````rust
let mut v: Vec<_> = ...
if v.is_empty() {
    v.sort(); // warn here
    borrow(&v);
} else {
    foo(&v);
}
````
